robottelo/hosts.py

```diff
@@ -107,6 +107,29 @@ def ip_addr(self):
         ipv4, ipv6 = self.execute('hostname -I').stdout.split()
         return ipv4
 
+    @cached_property
+    def distro_version(self):
+        """Get host's distribution information
+
+        :returns: A tuple in the form ``(distro, major, minor)``. ``major`` and
+            ``minor`` are integers. ``minor`` can be ``None`` if not available.
+
+        """
+        result = self.execute('cat /etc/redhat-release')
+        if result.status != 0:
+            raise ContentHostError(f'Not able to cat /etc/redhat-release "{result.stderr}"')
+        match = re.match(
+            r'(?P<distro>.+) release (?P<major>\d+)(.(?P<minor>\d+))?', result.stdout[0]
+        )
+        if match is None:
+            raise ContentHostError(f'Not able to parse release string "{result.stdout[0]}"')
+        groups = match.groupdict()
+        return (
+            groups['distro'],
+            int(groups['major']),
+            groups['minor'] if groups['minor'] is None else int(groups['minor']),
+        )
```

Comment:
Would this be easier if you used a `Version` instance instead? Even potentially returning a `Version` instance?

Reply:
oooooo yes it would. Want me to include that change in the scope? I'll have to update uses to reference the relevant attributes of the version instance instead of indexing.

Reply:
If the scope isn't too big..

Reply:
![image](https://user-images.githubusercontent.com/21315076/130697053-88ea1e3f-8b28-4f73-ad72-dad6877bd43b.png)

Reply:
Looks like this method, as copied, might not even work. Splitting the distro name out, since `packaging.version.Version` doesn't support such a field.

Reply:
Done, 009e1f7ef